core/src/main/java/org/jruby/util/Dir.java

```diff
@@ -993,14 +1001,11 @@ private static int glob_helper(Ruby runtime, String cwd, byte[] scheme,
                     for ( DirGlobber globber : links ) {
                         final ByteList link = globber.link;
                         if ( status == 0 ) {
-                            resource = JRubyFile.createResource(runtime, cwd, RubyString.byteListToString(link));
-                            if ( resource.isDirectory() ) {
-                                final int len = link.getRealSize();
-                                buf.length(0);
-                                buf.append(link);
-                                buf.append(path, SLASH_INDEX, end - SLASH_INDEX);
-                                status = glob_helper(runtime, cwd, scheme, buf, buf.getBegin() + len, flags, func, arg);
-                            }
+                            final int len = link.getRealSize();
+                            buf.length(0);
+                            buf.append(link);
+                            buf.append(path, SLASH_INDEX, end - SLASH_INDEX);
+                            status = glob_helper(runtime, cwd, scheme, buf, buf.getBegin() + len, flags, func, arg);
```

Comment:
so i moved the `isDirectory` check to where we append to `links` rather than the loop. This _might_ cause an issue  where links contains only non-directories, it should still call `break mainLoop`, although I haven't seen any issues. I don't know how I would cause this to happen.

Reply:
I've reverted this change